### PR TITLE
Fix Sonic 1's attract mode

### PR DESCRIPTION
**Pull request by bachimusprime**

### CODE REVIEW

File: emu/src/mem.rs [hunk 1]
```diff
@@ -51,16 +51,17 @@ impl Mem {
...
     pub fn write_u8(&mut self, address: u32, value: u32) {
         match address & 0xFFFFFF {
-            0..=0x3FFFFF => {},
+            0..=0x9FFFFF => {},
             0xA00000..=0xA03FFF => self.z80.write_byte(address, value),
             0xA04000..=0xA0FFFF => {},
-            0xA10000..=0xA1001F => self.io.write_byte(address, value),
-            0xA11100..=0xA112FF => self.z80.ctrl_write(address, value),
+            0xA10000..=0xA10FFF => self.io.write_byte(address & 0xFFF01F, value),
+            0xA11000..=0xA110FF => {},
```
Comment (kirjavascript) [inline emu/src/mem.rs:58]: this is part of the IO still so should be handled there instead (in io.write_byte)

File: emu/src/mem.rs [hunk 2]
```diff
@@ -51,16 +51,17 @@ impl Mem {
...
+            0xA11100..=0xA11FFF => self.z80.ctrl_write(address, value),
             0xC00000..=0xDFFFFF => {/* PSG goes here? */},
             0xFF0000..=0xFFFFFF => {
                 self.ram[address as usize & 0xFFFF] = value as u8;
             },
-            _ => todo!("write byte {:X} {:X}", address, value),
+            _ => {}, //todo!("write byte {:X} {:X}", address, value),
```
Comment (kirjavascript) [inline emu/src/mem.rs:64]: this should still work now without being commented out, right?

**Review by kirjavascript — CHANGES_REQUESTED** [review]

File: emu/src/mem.rs (hunk 2 above)
Comment (bachimusprime) [reply emu/src/mem.rs:64]: Nope. It writes to addresses where I have no idea what they're supposed to do.

File: emu/src/mem.rs (hunk 1 above)
Comment (bachimusprime) [reply emu/src/mem.rs:58]: Well, the thing is that 0xA10000 is mirrored through at least 0xA100FF, so I don't know how exactly to handle that without more extensive changes to the code.

Comment (kirjavascript) [reply emu/src/mem.rs:58]: you could just do it without mirroring. do the same as here (nothing) but within that function instead

File: emu/src/mem.rs (hunk 2 above)
Comment (kirjavascript) [reply emu/src/mem.rs:64]: do you know which addresses these are? can we put them in the match statement and add a comment about them instead? I would like to keep this 'catch all' section to panic on unimplemented things still

File: emu/src/mem.rs (hunk 1 above)
Comment (bachimusprime) [reply emu/src/mem.rs:58]: I guess, yeah

File: emu/src/mem.rs (hunk 2 above)
Comment (bachimusprime) [reply emu/src/mem.rs:64]: I forget honestly, since I was hitting a TON of them, so I just decided to comment that out lol

**Comment by kirjavascript** [comment]
s1 works now, and this has conflicts

**Pull request closed without merging**